x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheDirectory.java

```diff
@@ -274,6 +278,7 @@ void writeCacheFile(FileChannel fc, long start, long end) throws IOException {
             int bytesCopied = 0;
             try (IndexInput input = in.openInput(cacheFileReference.getFileName(), ioContext)) {
                 stats.incrementInnerOpenCount();
+                final long startTimeNanos = currentTimeNanosSupplier.getAsLong();
```

Comment:
Interestingly, this captures the start time after the initial blob read request has been made. I'm wondering if we should capture it before the `openInput()` which sends the underlying S3 request and update the stats in a finally block?

Reply:
I'm perhaps missing something here, because I think that we don't interact with the blob store until the first `readBytes` call. We certainly shouldn't start reading before we've called `seek` since seeks mean we must discard some work. Are you sure we do anything worth timing in `openInput()`?

Reply:
Oh no you're perfectly right, I confused the `openInput()` method with the `readBlob()` :man_facepalming:  Sorry for the noise.